core/modules/cart/store/actions.ts

```diff
@@ -148,239 +114,173 @@ const actions: ActionTree<CartState, RootState> = {
-              _connectBypassCount = _connectBypassCount + 1
-              await dispatch('connect', { guestCart: true })
-              Logger.error(task.result, 'cart')()
-            }
-          }
-        })
-        return diffLog
-      } else {
-        return diffLog
-      }
-    } else {
-      return diffLog
+  async sync ({ getters, rootGetters, commit, dispatch }, { forceClientState = false, dryRun = false }) {
+    const shouldUpdateClientState = rootGetters['checkout/isUserInCheckout'] || forceClientState
+    const { getCartItems, canUpdateMethods, isSyncRequired, bypassCounter } = getters
+    if (!canUpdateMethods || !isSyncRequired) return
+
+    commit(types.CART_SET_SYNC)
+    const { result, resultCode } = await CartService.pullCart()
+    if (resultCode === 200) {
+      return dispatch('merge', {
+        dryRun,
+        serverItems: result,
+        clientItems: getCartItems,
+        forceClientState: shouldUpdateClientState
+      })
+    }
+
+    if (bypassCounter < config.queues.maxCartBypassAttempts) {
+      Logger.log('Bypassing with guest cart' + bypassCounter, 'cart')()
+      commit(types.CART_UPDATE_BYPASS_COUNTER, { counter: 1 })
+      await dispatch('connect', { guestCart: true })
     }
+
+    Logger.error(result, 'cart')
   },
   /** @deprecated backward compatibility only */
   async serverPull ({ dispatch }, { forceClientState = false, dryRun = false }) {
     Logger.warn('The "cart/serverPull" action is deprecated and will not be supported with the Vue Storefront 1.11', 'cart')()
     return dispatch('sync', { forceClientState, dryRun })
   },
-  /** @description this method is part of "public" cart API */
-  async load ({ getters, commit, rootGetters, dispatch }, { forceClientState = false }: {forceClientState?: boolean} = {}) {
-    if (isServer) return
-    const cartShippingMethod = getters.getShippingMethod
-    if ((!cartShippingMethod || !cartShippingMethod.method_code) && (Array.isArray(rootGetters['shipping/shippingMethods']))) {
-      let shippingMethod = rootGetters['shipping/shippingMethods'].find(item => item.default)
-      commit(types.CART_UPD_SHIPPING, shippingMethod)
+  async setDefaultCheckoutMethods ({ getters, rootGetters, commit }) {
+    if (!getters.getShippingMethodCode) {
+      commit(types.CART_UPD_SHIPPING, rootGetters['shipping/getDefaultShippingMethod'])
     }
-    const cartPaymentMethpd = getters.getPaymentMethod
-    if ((!cartPaymentMethpd || !cartPaymentMethpd.code) && Array.isArray(rootGetters['payment/paymentMethods'])) {
-      let paymentMethod = rootGetters['payment/paymentMethods'].find(item => item.default)
-      commit(types.CART_UPD_PAYMENT, paymentMethod)
+
+    if (!getters.getPaymentMethodCode) {
+      commit(types.CART_UPD_PAYMENT, rootGetters['payment/getDefaultPaymentMethod'])
     }
+  },
+  async synchronizeCart ({ commit, dispatch }, { forceClientState }) {
+    const { synchronize, serverMergeByDefault } = config.cart
```

Comment:
We should keep “load” - it loads cart from localStorage sync is a side effect plus backward compatibility - this action is widely used across the themes so users lost the upgrade option with this change for not much value in exchange.

Action names shouldn’t be changed in cart module

Reply:
@pkarw 

> Action names shouldn’t be changed in cart module

Yeah, I know about it, this action is a new one, old load is here:
https://github.com/DivanteLtd/vue-storefront/pull/3317/files/f245896893f359f924049b6f3a0ba42c18ff0f95#diff-2101c92c40c144e3e56fb400ac211e3dR208
I will not change name of any old action, I just create a new ones when some actions are too complex

Reply:
OK, fair enough